### PR TITLE
Bump bbtools version and expand bbmap params

### PR DESCRIPTION
**Pull request by gregvonkuster**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/bbtools/macros.xml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <macros>
     <token name="@WRAPPER_VERSION@">1.0.0</token>
-    <token name="@VERSION_SUFFIX@">2</token>
-    <token name="@PROFILE@">20.09</token>
+    <token name="@VERSION_SUFFIX@">3</token>
+    <token name="@PROFILE@">21.09</token>
```
Comment (bgruening) [inline tools/bbtools/macros.xml:4]: is this profile increaes needed? You will exclude a few older Galaxy releases.

File: tools/bbtools/bbmap.xml [hunk 2]
```diff
@@ -44,59 +44,274 @@
...
+quickmatch='$output_options.quickmatch'
+trimreaddescriptions='$output_options.trimreaddescriptions'
+machineout='$output_options.machineout'
+printunmappedcount='$output_options.printunmappedcount'
+renamebyinsert='$output_options.renamebyinsert'
+#if str($output_options.output_sort) == 'coordinate':
+    out='all_reads.bam'
+    outu='unmapped_reads.bam'
+    outm='mapped_reads.bam'
+    ; samtools sort -@\${GALAXY_SLOTS:-4} -T "\${TMPDIR:-.}" -O bam -o '$output_all_reads' 'all_reads.bam'
```
Comment (bgruening) [inline tools/bbtools/bbmap.xml:140]: please use && instead of ;

**Review by bgruening — COMMENTED** [review]
Do you expect that the BAM files become smaller?

File: tools/bbtools/macros.xml (hunk 1 above)
Comment (gregvonkuster) [reply tools/bbtools/macros.xml:4]: I thought the general rule was to not have the profile release more than 12 months prior to the current release, but I am not sure.  Please let me know what it should be.

File: tools/bbtools/bbmap.xml (hunk 2 above)
Comment (gregvonkuster) [reply tools/bbtools/bbmap.xml]: Done - thanks!

File: tools/bbtools/macros.xml (hunk 1 above)
Comment (bgruening) [reply tools/bbtools/macros.xml:4]: Not sure if we have a rule, but this here is just 3 month :)

**Comment by gregvonkuster** [comment]
> Do you expect that the BAM files become smaller?

I'm not sure what you're asking here.  Are you referring to the delta values that I've put on the test outputs?

**Comment by bgruening** [comment]
This one here: https://github.com/galaxyproject/tools-iuc/pull/4379/files#diff-64bbbf6a4a6ec5d69194cf1bde0afe22fc3f2d9d5b47457cafc616394789e268

File: tools/bbtools/macros.xml (hunk 1 above)
Comment (gregvonkuster) [reply tools/bbtools/macros.xml:4]: Ok, I'll switch it back - thanks!

**Comment by gregvonkuster** [comment]
> This one here: https://github.com/galaxyproject/tools-iuc/pull/4379/files#diff-64bbbf6a4a6ec5d69194cf1bde0afe22fc3f2d9d5b47457cafc616394789e268

I don't think bam file sizes will necessarily change with the new release, but I haven't read all of the release notes.

File: tools/bbtools/macros.xml (hunk 1 above)
Comment (gregvonkuster) [reply tools/bbtools/macros.xml]: This has been reverted - thanks!

**Review by bgruening — APPROVED** [review]

**Comment by gregvonkuster** [comment]
Merci!  :)

**Pull request merged**